lib/commons/aria/get-element-unallowed-roles.js

```diff
@@ -0,0 +1,85 @@
+/* global aria */
+/**
+ * gets all unallowed roles for a given node
+ * @method getElementUnallowedRoles
+ * @param {Object} node HTMLElement to validate
+ * @param {String} tagName tag name of a node
+ * @param {String} allowImplicit option to allow implicit roles, defaults to true
+ * @return {Array<String>} retruns an array of roles that are not allowed on the given node
+ */
+aria.getElementUnallowedRoles = function getElementUnallowedRoles(
+	node,
+	allowImplicit
+) {
+	/**
+	 * Get roles applied to a given node
+	 * @param {HTMLElement} node HTMLElement
+	 * @return {Array<String>} return an array of roles applied to the node, if no roles, return an empty array.
+	 */
+	// TODO: not moving this to outer namespace yet, work with wilco to see overlap with his PR(WIP) - aria.getRole
+	function getRoleSegments(node) {
+		let roles = [];
+		if (!node) {
+			return roles;
+		}
+		if (node.hasAttribute('role')) {
+			const nodeRoles = axe.utils.tokenList(
+				node.getAttribute('role').toLowerCase()
+			);
+			roles = roles.concat(nodeRoles);
+		}
+		if (node.hasAttributeNS('http://www.idpf.org/2007/ops', 'type')) {
+			const epubRoles = axe.utils
+				.tokenList(
+					node
+						.getAttributeNS('http://www.idpf.org/2007/ops', 'type')
+						.toLowerCase()
+				)
+				.map(role => `doc-${role}`);
+			roles = roles.concat(epubRoles);
+		}
+		return roles;
+	}
+
+	const tagName = node.nodeName.toUpperCase();
+
+	// by pass custom elements
+	if (!axe.utils.isHtmlElement(node)) {
+		return [];
+	}
+
+	// by pass other elements that may not be a known html element
+	if (
+		document.createElement(tagName).toString() === '[object HTMLUnknownElement]'
```

Comment:
Unnecessary due to line 47. Please remove.

Reply:
Custom components that are written to override generic html tags still do not change the type of tag. Tested here - https://jsfiddle.net/Jey/bprv8fqe/7/

Hence removing this check.